### PR TITLE
types 7 - more menus!

### PR DESCRIPTION
**Pull request by hawkrives**
GitHubMenu and BonAppMenu.

Thoughts?

```diff
diff -wu tsc-errors.main tsc-errors.branch
--- tsc-errors.main	2022-04-20 15:35:33.000000000 -0400
+++ tsc-errors.branch	2022-04-20 15:35:22.000000000 -0400
@@ -160,13 +160,7 @@
 source/views/menus/lib/process-menu-shorthands.ts(25,3): error TS2739: Type '{}' is missing the following properties from type 'MonotonyContainer': id, name, image
 source/views/menus/lib/process-menu-shorthands.ts(31,3): error TS2740: Type '{}' is missing the following properties from type 'NutritionDetailContainer': calories, servingSize, fatContent, saturatedFatContent, and 7 more.
 source/views/menus/lib/process-menu-shorthands.ts(35,3): error TS2783: 'special' is specified more than once, so this usage will be overwritten.
-source/views/menus/menu-bonapp.tsx(142,8): error TS2571: Object is of type 'unknown'.
-source/views/menus/menu-bonapp.tsx(145,41): error TS2571: Object is of type 'unknown'.
-source/views/menus/menu-bonapp.tsx(159,25): error TS7006: Parameter 'grouped' implicitly has an 'any' type.
-source/views/menus/menu-bonapp.tsx(200,26): error TS7006: Parameter 's' implicitly has an 'any' type.
-source/views/menus/menu-bonapp.tsx(271,30): error TS2339: Property 'cafe' does not exist on type 'never'.
-source/views/menus/menu-bonapp.tsx(294,28): error TS2339: Property 'cor_icons' does not exist on type 'never'.
-source/views/menus/menu-github.tsx(103,5): error TS2741: Property 'cafeMessage' is missing in type '{ foodItems: MenuItemContainerType; meals: ProcessedMealType[]; menuCorIcons: MasterCorIconMapType; name: string; navigation: NavigationScreenProp<...>; now: Moment; }' but required in type 'ReactProps'.
+source/views/menus/menu-github.tsx(47,14): error TS2571: Object is of type 'unknown'.
 source/views/news/index.tsx(7,18): error TS2554: Expected 2 arguments, but got 1.
 source/views/news/index.tsx(9,13): error TS7031: Binding element 'navigation' implicitly has an 'any' type.
 source/views/news/index.tsx(13,27): error TS2339: Property 'stolaf' does not exist on type 'typeof import("/Users/rives/Projects/allaboutolaf/images/news-sources/index")'.
```

### CODE REVIEW

File: source/views/menus/menu-github.tsx [hunk 1]
```diff
@@ -1,113 +1,107 @@
...
+	useEffect(() => {
+		;(async () => {
+			setloading(true)
 
-		let container = await fetch(API('/food/named/menu/the-pause')).json<{
-			data: {
-				foodItems: MenuItemType[]
-				stationMenus: StationMenuType[]
-				corIcons: MasterCorIconMapType
+			let container
```
Comment (rye) [inline source/views/menus/menu-github.tsx]: Can we give this a type annotation?

Comment (hawkrives) [reply source/views/menus/menu-github.tsx]: That's probably a better spot for it. Sure thing.

File: source/views/menus/menu-bonapp.tsx [hunk 2]
```diff
@@ -70,234 +59,219 @@ function findCafeMessage(cafeInfo: CafeInfoType, now: Moment): string | null {
...
-				now: moment.tz(timezone()),
-			}))
+			setCafeMenu(cafeMenu ?? null)
+			setCafeInfo(cafeInfo ?? null)
+			setNow(moment.tz(timezone()))
 		} catch (error) {
+			if (!(error instanceof Error)) {
+				setErrorMessage('Unknown Error: Not an Error')
+				return
+			}
```
Comment (drewvolz) [inline source/views/menus/menu-bonapp.tsx:211]: This is an interesting error case?

**Review by drewvolz — APPROVED** [review]

File: source/views/menus/menu-github.tsx (hunk 1 above)
Comment (hawkrives) [reply source/views/menus/menu-github.tsx]: Done!

File: source/views/menus/menu-bonapp.tsx (hunk 2 above)
Comment (hawkrives) [reply source/views/menus/menu-bonapp.tsx:211]: Typescript treats a catch statement as `catch (err: unknown|any)`, so it complains that it doesn't know about the "message" property… so you have to check for instanceof Error first. 

Which technically makes sense - you can `throw` anything - but isn't a case I'd considered before.

**Pull request merged**